src/op_handlers/revert.rs

```diff
@@ -38,6 +38,12 @@ pub fn revert(vm: &mut VMState, opcode: &Opcode) -> Result<bool, EraVmError> {
         }
         Ok(false)
     } else {
+        let register = vm.get_register(opcode.src0_index);
+        let result = get_forward_memory_pointer(register.value, vm, register.is_pointer)?;
+        vm.set_register(
+            opcode.src0_index,
+            TaggedValue::new_pointer(FatPointer::encode(&result)),
+        );
```

Comment:
We use the same logic in three places, maybe it's time to define a specific function for this. If you don't think it's necessary, I'm fine with that

Reply:
Adding to this, it may be useful to change:
`if vm.running_contexts.len() > 1 || !vm.current_context()?.near_call_frames.is_empty()`
to:
`if vm.in_far_call() || vm.in_near_call()`
for better clarity across all places where this checks may appear.

Not related to this PR, but might as well add it here